src/features/cart.js

```diff
@@ -101,6 +101,20 @@ class Cart {
     });
   }
 
+  /**
+   * Check whether the no. of items in cart are available
+   * @param {string} id
+   * @param {number} quantity
+   * @returns {Promise}
+   */
+
+  checkQuantity(id, quantity) {
+    return this.commerce.request(`products/${id}`).then(response => ({
+      is_available: quantity <= response.quantity,
+      quantity: response.quantity,
+    }));
+  }
```

Comment:
```suggestion
  /**
   * Check whether the specified quantity is in stock/available for the specified product ID.
   *
   * @param {string} productId
   * @param {number} quantity
   * @returns {Promise}
   */
  checkQuantity(productId, quantity) {
    return this.commerce.request(`products/${id}`)
      .then(response => (quantity <= response.quantity));
  }
```

Untested, but I'd prefer it if the promise resolves to a boolean indicating whether the requested quantity is available for the product ID.

Reply:
I think for most of the scenerios, we'll need quantity too. It'll be helpful for developers to provide a reference to the users about the availability of any product.

Reply:
Which scenarios do you see this being used? I'm thinking that for displaying products in a grid, with the number that are in stock, we already have that data from the "get/list products" API. If you wanted to use this helper method to check if the quantity is available when you're adding a product to your cart, then we only care about whether it's acceptable or not.

Are there other scenarios I'm missing? If so, I'm happy to include the product's stock level in the response, but if there's no reason to include it then I'm inclined to remove it and simply return a boolean.

Reply:
Yeah, it sounds reasonable to me.